src/hugr/views/sibling_subgraph.rs

```diff
@@ -854,17 +890,21 @@ mod tests {
 
     #[test]
     fn non_convex_subgraph() {
-        let (hugr, func_root) = build_hugr().unwrap();
+        let (hugr, func_root) = build_3not_hugr().unwrap();
         let func: SiblingGraph<'_> = SiblingGraph::try_new(&hugr, func_root).unwrap();
-        let (inp, out) = hugr.children(func_root).take(2).collect_tuple().unwrap();
-        let first_cx_edge = hugr.node_outputs(inp).next().unwrap();
-        let snd_cx_edge = hugr.node_inputs(out).next().unwrap();
-        // All graph but one edge
+        let (inp, _out) = hugr.children(func_root).take(2).collect_tuple().unwrap();
+        let not1 = hugr.output_neighbours(inp).exactly_one().unwrap();
+        let not2 = hugr.output_neighbours(not1).exactly_one().unwrap();
+        let not3 = hugr.output_neighbours(not2).exactly_one().unwrap();
+        let not1_inp = hugr.node_inputs(not1).next().unwrap();
+        let not1_out = hugr.node_outputs(not1).next().unwrap();
+        let not3_inp = hugr.node_inputs(not3).next().unwrap();
+        let not3_out = hugr.node_outputs(not3).next().unwrap();
         assert!(matches!(
             SiblingSubgraph::try_new(
-                vec![vec![(out, snd_cx_edge)]],
-                vec![(inp, first_cx_edge)],
-                &func,
+                vec![vec![(not1, not1_inp)], vec![(not3, not3_inp)]],
+                vec![(not1, not1_out), (not3, not3_out)],
```

Comment:
The updated test is definitely a good example of non-convex subgraph. The old test was more of the pathological kind. But it should still fail the (updated) convexity check, doesn't it?

If possible it would be good to keep the old test too. Another, slightly less pathological but also non-convex subgraph should be
```rust
        let (hugr, func_root) = build_hugr().unwrap();
        let func: SiblingGraph<'_> = SiblingGraph::try_new(&hugr, func_root).unwrap();
        let (inp, out) = hugr.children(func_root).take(2).collect_tuple().unwrap();
        let cx_edges_in = hugr.node_outputs(inp);
        let cx_edges_out = hugr.node_inputs(out);
        // All graph but the CX
        assert!(matches!(
            SiblingSubgraph::try_new(
                cx_edges_out.map(|p| vec![(out, p)]).collect(),
                cx_edges_in.map(|p| (in, p)).collect(),
                &func,
            ),
            Err(InvalidSubgraph::NotConvex)
        ));
```

Does this make sense (I didnt run it)?

Reply:
> Does this make sense (I didnt run it)?

Yes I'll add this test, but it produces `InvalidBoundary` rather than `NotConvex`. Is this what you would expect?